### ISSUE TITLE
The first authentication isn't working on iPad devices

### ISSUE THREAD

**Issue by Mattzard** [aspnet-contrib/AspNet.Security.OAuth.Providers#278]
I have tried a number of iPad devices here, and it seems to fail. See the scenario below:

1. On an iPad, completely clear your cookies/session or open an Incognito session
2. Now you have a clean session, try to login via LinkedIn/Microsoft using this library. 
3. You will be taken to LinkedIn/Microsoft's login page where you should successfully enter your credentials

At this point, you should notice you are returned to your application yet the `User.Claims` is empty. I have also checked `User.Identity.IsAuthenticated` and this is showing as False.

If you again try to connect via the same social provider as in step 2, you will see that the returned `User.Claims` is now populated as expected. It seems you must be logged into the social provider's website first for this to be possible and work as expected. 

I have also tried, from step 1 above, to log into the social provider's website first, before completing step 2, and I was able to login without any issues. 

Is anyone else having this issue for iPad devices? I'm struggling to find a solution to this which is why I have posted my question here, but I will continue to investigate and update if I find anything, or if it is indeed my mistake.

Thanks for the help.

**Comment by martincostello** [comment]
It's possible that the issue you're seeing is due to this:

https://github.com/aspnet/Announcements/issues/318

**Comment by Mattzard** [comment]
Apologies, I forgot to add details to this when I discovered what the problem was. 

I found out the problem occurred because I had a Mobile App installed on my Phone that was running against the same domain as my API that contained the code to allow for social login that this github project provides access to. 

Simply changing the domain of my API allowed the social login code to work perfectly on iPad devices.

Sorry for not updating sooner!
